### ISSUE TITLE
Embed into Webpage

### ISSUE THREAD

**Issue by Zach-15803** [shellinabox/shellinabox#435]
So Here's my dilemma, i have a website I'm working on, and for example purposes, ill refer to it as mydomain.com

I have SIAB (Shell in a Box) configured to port 8022, so if i were to go to it, i would need to go to mydomain.com:8022
but i want to be able to either Embed it in a Web-page, or have my Apache configured so that mydomain.com:8022 = mydomain.com/shell

any ideas? its a 32bit Ubuntu 16.04 server with 64gb of storage (VM)

I want it to do this because the website hosting service only allows 1 port to be used, (serveo.net) at a time if i want one URL to remember, which I only want 1 URL to remember.

On my server i had at my school i had something like this set up using VirtualHost configurations in apace, but i have since forgotten how to do that and School is out for the summer so i cant retrieve the configuration file.

**Comment by arnowelzel** [comment]
You may a proxy redirect in .htaccess:

RewriteEngine On
RewriteRule ^shell/(.*)$ http://mydomain.com:8022/$1 [P,L]

A reverse proxy setup should do as well.

**Comment by uudruid74** [comment]
@arnowelzel He said he can only have 1 port open.

@Zach-15803 You want to set up your webserver to proxy a specific URL (like the example given by arnowelzel) to localhost:8022.  A rewrite won't do it.  Add a location directive, such as ..

<Location /shell>

    ProxyPass http://127.0.0.1:8022

</Location>

**Comment by uudruid74** [comment]
Thanks for that info. I stand corrected. I switch to nginx a while back and haven't really been following up on all the new Apache trans. Thanks

[Evan Langlois - Contact Using Hop](http://GetHop.com/?_hmid=1531690779)

On July 15, 2018 at 21:25 GMT, Arno Welzel <notifications@github.com> wrote:

[@uudruid74](https://github.com/uudruid74) In fact I use this on my server exactly this way. See here about RewriteRule using the [P] flag:

https://httpd.apache.org/docs/2.4/rewrite/flags.html#flag_p

Use of the [P] flag causes the request to be handled by mod_proxy, and handled via a proxy request.

However, this requires Apache 2.4 and for Apache 2.2 your solution is another way to achieve this result.

**Comment by Zach-15803** [comment]
Thanks, this is useful to make server management a lot easier when outbound connections on port 22 are blocked by school wifi
